backend/src/app.module.ts

```diff
@@ -8,10 +8,16 @@ import { LitService } from './lit/lit.service';
 import { CryptionService } from './cryption/cryption.service';
 import { IpfsApiService } from './ipfs-api/ipfs-api.service';
 import { HttpModule } from '@nestjs/axios';
+import { RegisterModule } from './register/register.module';
+import { RegisterController } from './register/register.controller';
+import { RegisterService } from './register/register.service';
+import { ClientSignModule } from './client-sign/client-sign.module';
+import { ClientSignController } from './client-sign/client-sign.controller';
+import { ClientSignService } from './client-sign/client-sign.service';
 
 @Module({
-    imports: [ConfigModule.forRoot(), IpfsApiModule, CryptionModule, HttpModule],
-    controllers: [AppController],
-    providers: [AppService, LitService, CryptionService, IpfsApiService],
+    imports: [ConfigModule.forRoot(), IpfsApiModule, CryptionModule, HttpModule, RegisterModule, ClientSignModule],
+    controllers: [AppController, RegisterController, ClientSignController],
```

Comment:
You are importing modules and adding the same controllers and providers (Register and ClientSign). This may result in multiple instances in the app. Instead, use the export in the module to export the shared Providers. Also, Controllers don't need to be imported here.

Reply:
Done